src/containers/stage.jsx

```diff
@@ -186,18 +186,18 @@ class Stage extends React.Component {
             mouseDown: false,
             mouseDownPosition: null
         });
+        const data = {
+            isDown: false,
+            x: x - this.rect.left,
+            y: y - this.rect.top,
+            canvasWidth: this.rect.width,
+            canvasHeight: this.rect.height
+        };
         if (this.state.isDragging) {
             this.onStopDrag();
-        } else {
-            const data = {
-                isDown: false,
-                x: x - this.rect.left,
-                y: y - this.rect.top,
-                canvasWidth: this.rect.width,
-                canvasHeight: this.rect.height
-            };
-            this.props.vm.postIOData('mouse', data);
+            data.wasDragged = true;
```

Comment:
This might be simpler to read (and more intuitive with people's understanding of `const`) if it was just part of the `data` object:
```javascript
        const data = {
            isDown: false,
            x: x - this.rect.left,
            y: y - this.rect.top,
            canvasWidth: this.rect.width,
            canvasHeight: this.rect.height,
            wasDragged: this.state.isDragging
        };
```